### DISCUSSION TITLE
Anomalib with Intel ARC GPU

### DISCUSSION THREAD

**Discussion by Nakagawatokuji** [open-edge-platform/anomalib#2243] [Q&A]
I am using an Intel ARC A310 graphics card. I installed Anomalib and tried some models, but it doesn't recognize my GPU and I can't utilize it. It runs on CPU.
I tried to install `Intel Extension for Pytorch'. But Anomalib installer seems to overwrite it. 
Does anyone have solutions? Thanks in advance.

**Comment by samet-akcay** [comment]
As far as I remember, @blaz-r tried it, and got it working. He can comment better.

Meanwhile, you could first install anomalib dependencies and install intel extension later after.

**Comment by blaz-r** [comment]
I didn't try to use it with Anomalib, but I use it to train my own model which does use some Anomalib things. You'll need to install Intel extension for pytorch, and verify that it works. As Samet said, to do that first install Anomalib dependencies and then install the Intel extension with the correct torch. 

I believe that once you have that sorted, all you have to do is import the extension (only in the main file) right after the torch like so:
```
import torch
import intel_extension_for_pytorch as ipex
``` 
and then specify the accelerator in Engine args as `xpu`. If you also have a iGPU inside the cpu, you'll probably need to make that `xpu:1`.

**Reply by samet-akcay** [reply]
There are PRs to integrate the Intel version of torch to the upstream torch and lightning. Once it's merged will be able to support it out of the box

**Comment by paularamo** [comment]
@Nakagawatokuji  thanks for opening this discussion. For running the inference on Intel GPUs, please refer to the OpenVINO Inferencer in Anomalib and follow the installation process Intel GPU Plugin for OpenVINO. https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/README.md, Once you have the GPU working with OpenVINO won't be blockers for running the Anomalib inference in your ARC. Best regards

**Reply by Nakagawatokuji** [reply]
Thank you all for the useful information!I will try again, checking based on the information you provided.
